clustered/client/src/main/java/org/ehcache/clustered/client/internal/store/operations/BaseOperation.java

```diff
@@ -16,26 +16,140 @@
 
 package org.ehcache.clustered.client.internal.store.operations;
 
+import org.ehcache.clustered.client.internal.store.operations.codecs.CodecException;
+import org.ehcache.spi.serialization.Serializer;
+
+import java.nio.ByteBuffer;
+
 /**
  * Base class that represents a {@link org.ehcache.Cache} operation
  *
  * @param <K> key type
  */
-public abstract class BaseOperation<K> implements Operation<K> {
+public abstract class BaseOperation<K, V> implements Operation<K, V> {
+
+  public static final int BYTE_SIZE_BYTES = 1;
+  public static final int INT_SIZE_BYTES = 4;
+  public static final int LONG_SIZE_BYTES = 8;
 
-  protected final K key;
+  protected K key;
+  protected V value;
+
+  public BaseOperation() {
+    this.key = null;
+    this.value = null;
+  }
 
   public BaseOperation(final K key) {
+    this(key, null);
```

Comment:
That's up-to the subclass. This is the constructor that `RemoveOperation` uses and to avoid that NPE it extends the _checkValueNonNull_ method. This is exactly the reason why I made this method protected so that sub-classes that don't care about the values can use this constructor and extend the _checkValueNonNull_ method to avoid NPE.

The other choice that I had was to keep the _checkValueNonNull_ in the base class do nothing and the sub-classes that requires a non-null value extend it in such a way that it throws. But then I'd have had to duplicate that error throwing logic to 5 different classes: PUT, PUT_IF_ABSENT, REMOVE_CONDITIONAL, REPLACE, REPLACE_CONDITIONAL. So I went for minimal codu duplication and did it this way. I understand that it wasn't a LOT of duplicated code. But still tried to avoid as much as I can.